mangaki/mangaki/utils/dpp.py

```diff
@@ -1,128 +1,48 @@
-
     def __init__(self, items):
         self.items = items
 
     def sample_k(self, nb_points):
-        return choice(self.items, nb_points).tolist()
+        return sample(self.items, nb_points)
 
 
 class MangakiDPP:
-
-    def __init__(self, items, similarity_matrix):
-        self.items = items
-        self.similarity_matrix = similarity_matrix
-
-    def sample_k(self, *args, **kwargs):
-        for i in range(MAX_ITER_SAMPLE_DPP):
-            try:
-                return self._sample_k(*args, **kwargs)
-            except np.linalg.linalg.LinAlgError as e:
-                print('LinAlgError in MangakiDPP')
-        raise ValueError('Too much LinAlgError')
-
-    def _sample_k(self, k, max_nb_iterations=1000, rng=np.random):
-        """
-        Thanks to mehdidc on github : https://github.com/mehdidc/dpp
-        Sample a list of k items from a DPP defined
-        by the similarity matrix L. The algorithm
-        is iterative and runs for max_nb_iterations.
-        The algorithm used is from
-        (Fast Determinantal Point Process Sampling with
-        Application to Clustering, Byungkon Kang, NIPS 2013)
-        """
-        items = self.items
-        L = self.similarity_matrix
-        initial = rng.choice(range(len(items)), size=k, replace=False)
-        X = [False] * len(items)
-        for i in initial:
-            X[i] = True
-        X = np.array(X)
-        for i in range(max_nb_iterations):
-            u = rng.choice(np.arange(len(items))[X])
-            v = rng.choice(np.arange(len(items))[~X])
-            Y = X.copy()
-            Y[u] = False
-            L_Y = L[Y, :]
-            L_Y = L_Y[:, Y]
-            L_Y_inv = np.linalg.inv(L_Y)
-            c_v = L[v:v + 1, :]
-            c_v = c_v[:, v:v + 1]
-            b_v = L[Y, :]
-            b_v = b_v[:, v:v + 1]
-            c_u = L[u:u + 1, :]
-            c_u = c_u[:, u:u + 1]
-            b_u = L[Y, :]
-            b_u = b_u[:, u:u + 1]
-            p = min(1, c_v - np.dot(np.dot(b_v.T, L_Y_inv), b_v) /
-                    (c_u - np.dot(np.dot(b_u.T, L_Y_inv.T), b_u)))
-            if rng.uniform() <= p:
-                X = Y[:]
-                X[v] = True
-        return np.array(items)[X]
-
-
-def compare(similarity, algos, nb_points, nb_iterations=20):
-
-    resultats = np.zeros([len(algos), 2])
-
-    for _ in range(nb_iterations):
-        for i in range(len(algos)):
-            items = algos[i].sample_k(nb_points)
-            points = similarity.matrix[:, items].T.toarray()
-
-            det = np.linalg.det(squareform(pdist(
-                points,
-                metric='cosine')))
-
-            diam = diameter_0(points)
-
-            resultats[i, 0] += det
-            resultats[i, 1] += diam
-    resultats /= nb_iterations
-
-    return resultats
+    def __init__(self, work_ids=None, vectors=None):
+        self.work_ids = np.array(work_ids)
+        self.set_vectors(vectors)
+
+    def set_vectors(self, vectors):
```

Comment:
Why do you need this function?

Doing `self.vectors = vectors` instead of calling it is faster and better, right?